code/lib/codemod/src/transforms/mdx-to-csf.ts

```diff
@@ -37,18 +40,35 @@ export default async function jscodeshift(info: FileInfo) {
     baseName += '_';
   }
 
-  const result = await transform(info, path.basename(baseName));
+  try {
+    const result = await transform(info, path.basename(baseName));
+
+    if (result[1] != null) {
+      fs.writeFileSync(`${baseName}.stories.js`, result[1]);
+    }
 
-  const [mdx, csf] = result;
+    renameList.push({ original: info.path, baseName });
 
-  if (csf != null) {
-    fs.writeFileSync(`${baseName}.stories.js`, csf);
+    return result[0];
+  } catch (e) {
+    brokenList.push({ original: info.path, baseName });
+    throw e;
   }
-
-  return mdx;
 }
 
-export async function transform(info: FileInfo, baseName: string): Promise<[string, string]> {
+process.on('exit', () => {
+  renameList.forEach((file) => {
+    fs.renameSync(file.original, `${file.baseName}.mdx`);
+  });
+  brokenList.forEach((file) => {
+    fs.renameSync(file.original, `${file.original}.broken`);
+  });
+});
+
```

Comment:
I don't have any in mind :/ Renaming during the transform will fail, because jscodeshift will complain that the file doesn't exist which it wants to transform. The solution is unconventional, but works. Do you have any concerns, where this solution might not work?